rust/lance-encoding/src/encoder.rs

```diff
@@ -773,9 +775,14 @@ impl CompressionStrategy for CoreArrayEncodingStrategy {
     fn create_miniblock_compressor(
         &self,
         field: &Field,
-        _data: &DataBlock,
+        data: &DataBlock,
     ) -> Result<Box<dyn MiniBlockCompressor>> {
         assert!(field.data_type().byte_width() > 0);
+        if let DataBlock::FixedWidth(ref fixed_width_data) = data {
+            if fixed_width_data.bits_per_value <= 64 {
```

Comment:
Should we also check that it is a multiple of `8`?  Or maybe that it is one of `[8, 16, 32, 64]`?  Would your logic work (for example) with `fixed-size-list<int8, 3>` which has a width of `24`?

Reply:
yeah, good idea. I should make sure it is one of `8, 16, 32, 64`.
one interesting issue may need to be considered:
when input is `fixed-size-list<int8, 4>`, we can bitpack it and get back the decoding result back correctly, but the compression ratio is probably bad, we may also need to make sure don't use bitpack when hit this case